From 8a4b0ef645f18a3de4697936bcd4b75b0b0549a2 Mon Sep 17 00:00:00 2001
From: Niklas Haas <git@haasn.xyz>
Date: Tue, 17 Oct 2017 15:48:28 +0200
Subject: [PATCH 21/22] vo_gpu: vulkan: omit needless #define

---
 video/out/gpu/shader_cache.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/video/out/gpu/shader_cache.c b/video/out/gpu/shader_cache.c
index 3b2578eae0..5789d54a11 100644
--- a/video/out/gpu/shader_cache.c
+++ b/video/out/gpu/shader_cache.c
@@ -796,11 +796,6 @@ static void gl_sc_generate(struct gl_shader_cache *sc,
         ADD(header, "#define texture texture2D\n");
     }
 
-    if (sc->ra->glsl_vulkan && type == RA_RENDERPASS_TYPE_COMPUTE) {
-        ADD(header, "#define gl_GlobalInvocationIndex "
-                    "(gl_WorkGroupID * gl_WorkGroupSize + gl_LocalInvocationID)\n");
-    }
-
     // Additional helpers.
     ADD(header, "#define LUT_POS(x, lut_size)"
                 " mix(0.5 / (lut_size), 1.0 - 0.5 / (lut_size), (x))\n");
-- 
2.14.3

