diff --git a/wscript b/wscript
index e25a7667ad..f355fd0f6b 100644
--- a/wscript
+++ b/wscript
@@ -462,14 +462,14 @@ libav_dependencies = [
         'req': True,
         'fmsg': "FFmpeg/Libav development files not found.",
     }, {
-        'name': 'is_ffmpeg_garbage',
+        'name': 'is_ffmpeg',
         'desc': 'libav* is upstream FFmpeg (unsupported)',
         # FFmpeg <=> LIBAVUTIL_VERSION_MICRO>=100
         'func': check_statement('libavcodec/version.h',
                                 'int x[LIBAVCODEC_VERSION_MICRO >= 100 ? 1 : -1]',
                                 use='libavcodec')
     }, {
-        'name': 'is_ffmpeg',
+        'name': 'is_mpv_ffmpeg',
         'desc': 'libav* is FFmpeg mpv modified version',
         'func': check_statement('libavcodec/version.h',
                                 'int x[LIBAVCODEC_MPV ? 1 : -1]',
@@ -487,7 +487,7 @@ libav_dependencies = [
     }, {
         'name': 'libav',
         'desc': 'Libav/FFmpeg library versions',
-        'deps': 'is_ffmpeg || is_libav',
+        'deps': 'is_ffmpeg || is_libav || is_mpv_ffmpeg',
         'func': check_ffmpeg_or_libav_versions(),
         'req': True,
         'fmsg': "Unable to find development files for some of the required \
