diff --git a/video/out/filter_kernels.c b/video/out/filter_kernels.c
index fe5265c..d779991 100644
--- a/video/out/filter_kernels.c
+++ b/video/out/filter_kernels.c
@@ -178,6 +178,12 @@ static double hamming(params *p, double x)
     return 0.54 + 0.46 * cos(M_PI * x);
 }
 
+// Generalization of raised-power cosine windows to arbitrary exponents
+static double petsu(params *p, double x)
+{
+    return powf(cos(M_PI/2 * x), p->params[0]);
+}
+
 static double quadric(params *p, double x)
 {
     if (x <  0.75) {
@@ -322,6 +328,7 @@ const struct filter_window mp_filter_windows[] = {
     {"bartlett",       1,   triangle},
     {"hanning",        1,   hanning},
     {"hamming",        1,   hamming},
+    {"petsu",          1,   petsu,    .params = {1.5, NAN} },
     {"quadric",        1.5, quadric},
     {"welch",          1,   welch},
     {"kaiser",         1,   kaiser,   .params = {6.33, NAN} },
