--- a/mozilla-release/browser/base/content/browser-places.js	2015-07-12 18:59:06.386767777 +0200
+++ b/mozilla-release/browser/base/content/browser-places.js	2015-07-12 18:59:19.938621549 +0200
@@ -1432,6 +1432,7 @@
 
   updatePocketItemVisibility: function BUI_updatePocketItemVisibility(prefix) {
     let hidden = !CustomizableUI.getPlacementOfWidget("pocket-button");
+    hidden = true; // Disable pocket
     if (!hidden) {
       let locale = Cc["@mozilla.org/chrome/chrome-registry;1"].
                    getService(Ci.nsIXULChromeRegistry).
