diff --git a/qutebrowser/browser/webkit/webpage.py b/qutebrowser/browser/webkit/webpage.py
index 7e1d991b9..fec6d0f49 100644
--- a/qutebrowser/browser/webkit/webpage.py
+++ b/qutebrowser/browser/webkit/webpage.py
@@ -366,11 +366,19 @@ class BrowserPage(QWebPage):
 
     def userAgentForUrl(self, url):
         """Override QWebPage::userAgentForUrl to customize the user agent."""
-        ua = config.val.content.headers.user_agent
+
+        special_agents = {
+            # Infinite redirect loop / garbage otherwise
+            'twitter.com': 'Mozilla/5.0 (compatible; Googlebot/2.1; +http://www.google.com/bot.html',
+        }
+
+        ua = special_agents.get(url.host())
         if ua is None:
-            return super().userAgentForUrl(url)
-        else:
-            return ua
+            ua = config.val.content.headers.user_agent
+        if ua is None:
+            ua = super().userAgentForUrl(url)
+
+        return ua
 
     def supportsExtension(self, ext):
         """Override QWebPage::supportsExtension to provide error pages.
