--- a/kdeui/kernel/kapplication.cpp	2015-06-26 05:14:18.000000000 +0200
+++ b/kdeui/kernel/kapplication.cpp	2015-08-07 17:02:22.895630076 +0200
@@ -512,12 +512,6 @@
   extern void qDBusBindToApplication();
   qDBusBindToApplication();
   QDBusConnectionInterface *bus = 0;
-  if (!QDBusConnection::sessionBus().isConnected() || !(bus = QDBusConnection::sessionBus().interface())) {
-      kFatal(240) << "Session bus not found" << endl <<
-                  "To circumvent this problem try the following command (with Linux and bash)" << endl <<
-                  "export $(dbus-launch)";
-      ::exit(125);
-  }
 
   extern bool s_kuniqueapplication_startCalled;
   if ( bus && !s_kuniqueapplication_startCalled ) // don't register again if KUniqueApplication did so already
