diff -ru a/coreutils-8.24/lib/human.c b/coreutils-8.24/lib/human.c
--- a/coreutils-8.24/lib/human.c	2015-06-16 07:00:37.000000000 +0200
+++ b/coreutils-8.24/lib/human.c	2015-08-13 04:38:32.432034227 +0200
@@ -153,6 +153,7 @@
 human_readable (uintmax_t n, char *buf, int opts,
                 uintmax_t from_block_size, uintmax_t to_block_size)
 {
+  n *= 8; // use bits instead of bytes
   int inexact_style =
     opts & (human_round_to_nearest | human_floor | human_ceiling);
   unsigned int base = opts & human_base_1024 ? 1024 : 1000;
@@ -382,6 +383,7 @@
         }
     }
 
+  *psuffix++ = 'b'; // bits
   *psuffix = '\0';
 
   return p;
