diff --git a/Source/WebKit/qt/WebCoreSupport/InspectorClientQt.cpp b/Source/WebKit/qt/WebCoreSupport/InspectorClientQt.cpp
index c809797..597ba95 100644
--- a/Source/WebKit/qt/WebCoreSupport/InspectorClientQt.cpp
+++ b/Source/WebKit/qt/WebCoreSupport/InspectorClientQt.cpp
@@ -145,6 +145,9 @@ InspectorClientQt::InspectorClientQt(QWebPageAdapter* page)
 
 void InspectorClientQt::inspectedPageDestroyed()
 {
+    if (m_frontendClient)
+        m_frontendClient->inspectorClientDestroyed();
+
     InspectorServerQt* webInspectorServer = InspectorServerQt::server();
     if (webInspectorServer)
         webInspectorServer->unregisterClient(this);