diff --git a/Source/WebCore/platform/network/qt/QNetworkReplyHandler.cpp b/Source/WebCore/platform/network/qt/QNetworkReplyHandler.cpp
index 69c7230..e943672 100644
--- a/Source/WebCore/platform/network/qt/QNetworkReplyHandler.cpp
+++ b/Source/WebCore/platform/network/qt/QNetworkReplyHandler.cpp
@@ -70,7 +70,11 @@ bool FormDataIODevice::reset()
         m_currentFile->close();
 
     m_currentDelta = 0;
-    m_formElements = m_formData->elements();
+
+    if (m_formData)
+        m_formElements = m_formData->elements();
+    else
+        m_formElements.clear();
 
     prepareCurrentElement();
     return true;
