diff --git a/src/nvim/memline.c b/src/nvim/memline.c
index 55e7e0182..de7727dc4 100644
--- a/src/nvim/memline.c
+++ b/src/nvim/memline.c
@@ -1744,7 +1744,7 @@ ml_get_buf (
       /* Avoid giving this message for a recursive call, may happen when
        * the GUI redraws part of the text. */
       ++recursive;
-      EMSGN(_("E315: ml_get: invalid lnum: %" PRId64), lnum);
+      //EMSGN(_("E315: ml_get: invalid lnum: %" PRId64), lnum);
       --recursive;
     }
 errorret:
@@ -1776,7 +1776,7 @@ errorret:
         /* Avoid giving this message for a recursive call, may happen
          * when the GUI redraws part of the text. */
         ++recursive;
-        EMSGN(_("E316: ml_get: cannot find line %" PRId64), lnum);
+        //EMSGN(_("E316: ml_get: cannot find line %" PRId64), lnum);
         --recursive;
       }
       goto errorret;
